pkg/setup/setup.go

```diff
@@ -136,7 +135,7 @@ func validateCanRunTargetArch(targetArch string) error {
 		// we could error here but in principle with a working qemu-user
 		// any arch should work so let's just warn. the common case
 		// (arm64/amd64) is covered properly
-		logrus.Warningf("cannot check architecture support for %v: no canary binary found", targetArch)
+		log.Panicf("cannot check architecture support for %v: no canary binary found", targetArch)
```

Comment:
Please do not change a warning into a panic in the middle of a refactor, if we want to do this, a change like this should be its own commit with a proper rational.